src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorAbsSendTime.java

```diff
@@ -0,0 +1,719 @@
+import org.jitsi.util.*;
+import org.jitsi.util.concurrent.RecurringRunnable;
+import org.jitsi.util.function.Predicate;
+
+import java.util.*;
+
+/**
+ * webrtc.org abs_send_time implementation as of June 26, 2017.
+ * commit ID: 23fbd2aa2c81d065b84d17b09b747e75672e1159
+ * @author Julian Chukwu
+ */
+public class RemoteBitrateEstimatorAbsSendTime
+    extends SinglePacketTransformer
+    implements RemoteBitrateEstimator,
+        RecurringRunnable
+{
+    private static final Logger logger = Logger
+            .getLogger(RemoteBitrateEstimatorAbsSendTime.class);
+
+    /**
+     * Ctor.
+     * <p>
+     * XXX At some point ideally we would get rid of this ctor and all the
+     * inheritors will use the parametrized ctor. Also, we might want to move
+     * this check inside the <tt>TransformEngineChain</tt> so that we only make
+     * the check once per packet: The RTCP transformer is only supposed to
+     * (reverse) transform RTCP packets and the RTP transformer is only
+     * supposed to modify RTP packets.
+     */
+    public RemoteBitrateEstimatorAbsSendTime() {
+        super();
+    }
+
+    /**
+     * Ctor.
+     *
+     * @param packetPredicate the <tt>PacketPredicate</tt> to use to match
+     *                        packets to (reverse) transform.
+     */
+    public RemoteBitrateEstimatorAbsSendTime(Predicate<ByteArrayBuffer> packetPredicate) {
+        super(packetPredicate);
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * The (default) implementation of {@code SinglePacketTransformer} does
+     * nothing.
+     */
+    @Override
+    public void close() {
+        super.close();
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * Reverse-transforms an array of packets by calling
+     * {@link #reverseTransform(RawPacket)} on each one.
+     *
+     * @param pkts
+     */
+    @Override
+    public RawPacket[] reverseTransform(RawPacket[] pkts) {
+        return super.reverseTransform(pkts);
+    }
+
+    /**
+     * Transforms a specific packet.
+     *
+     * @param pkt the packet to be transformed.
+     * @return the transformed packet.
+     */
+    @Override
+    public RawPacket transform(RawPacket pkt) {
+        return null;
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * Transforms an array of packets by calling {@link #transform(RawPacket)}
+     * on each one.
+     *
+     * @param pkts
+     */
+    @Override
+    public RawPacket[] transform(RawPacket[] pkts) {
+        return super.transform(pkts);
+    }
+
+    /**
+     * Returns the number of milliseconds until this instance wants a worker
+     * thread to call {@link #run()}. The method is called on the same
+     * worker thread as Process will be called on.
+     *
+     * @return the number of milliseconds until this instance wants a worker
+     * thread to call {@link #run()}
+     */
+    @Override
+    public long getTimeUntilNextRun() {
+        return 0;
+    }
+
+    private enum ProbeResult
```

Comment:
Please move class/enum definitions to the end of the file (after method definitions)